### PR TITLE
itest: add TxPublisher integration tests

### PR DESCRIPTION
**Pull request by Abdulkbk**
Implements Task 435 — cover transaction publication contracts.

`CheckMempoolAcceptance` and `Broadcast` had unit coverage against a mocked
chain and no integration coverage at all. This adds six cases in
`itest/tx_publisher_test.go`.

## Task 435 acceptance criteria mapped to tests

| Criterion | Case |
| --- | --- |
| Accepted spend | `txpublisher check acceptance`, `txpublisher broadcast transaction` |
| Representative policy rejection | `txpublisher reject acceptance` (dust → `chain.ErrDust`), `txpublisher reject broadcast` |
| Malformed or conflicting | `txpublisher reject acceptance` (unknown input → `chain.ErrMissingInputs`, duplicate input → `chain.ErrDuplicateInput`) |
| Already-known behavior | `txpublisher broadcast known` |
| Stopped Wallet state | `txpublisher wallet state` (both methods, before Start and after Stop; nil transaction on a running wallet) |
| Acceptance check never publishes | `txpublisher check acceptance` (mempool, wallet records and coins all unchanged) |
| Broadcast observable before and after mining | `txpublisher broadcast transaction` |
| kvdb, SQLite and PostgreSQL, no skip branch | Full suite passes on all three; no case contains database-conditional code |
| Test-only | No production file is touched |

## Harness changes

Both additive, in `bwtest`:

- `SpendFixture` and `SignSpend` — authors and signs a transaction from named
  wallet coins, using the inputs and outputs verbatim. Not built on
  `CreateTransaction`/`FundPsbt`, which enforce the creation-time policy Task
  434 owns and reject the dust fixture outright.
- `MempoolAcceptanceSupported` — one authority for which chain backends answer
  the acceptance check.

## Backend neutrality

The rejections were chosen so one identity holds on every backend. Three
candidates were measured and discarded:

- **Minimum relay fee** — btcd reports a mempool minimum, bitcoind a relay
  minimum. Dust carries the policy requirement instead.
- **Double spend** — bitcoind enables full replace-by-fee by default, so it
  judges the second transaction as a replacement bid, accepting it when it pays
  more and reporting an insufficient fee when it pays less, while btcd reports a
  mempool conflict. `MaxTxInSequenceNum` does not prevent this. Duplicate inputs
  cover the malformed requirement instead.
- **No outputs** — btcd rejects on serialized size before reaching the
  empty-output rule, mapping to `ErrUndefined`.

Neutrino has no mempool and reports the acceptance check as unimplemented. The
two cases whose subject is that check assert `chain.ErrUnimplemented` there
rather than skipping, so every backend gets a definite assertion. Everything
else is backend-neutral with no split, including the dust refusal, which
neutrino's peer returns over the wire.

Part of #1250.

### CODE REVIEW

File: itest/tx_publisher_test.go [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+
+// spendFee is the fee in satoshis that the publication fixtures leave behind.
+// It clears the relay minimum by a wide margin while staying far below the
+// backend's default maximum fee rate of 10,000 sat/vbyte, above which an
+// otherwise valid transaction is rejected as paying absurdly much.
+const spendFee = 10_000
+
+// dustAmount is below the relay's dust threshold for every output type the
+// wallet produces, so a payment of it is refused as dust.
+const dustAmount = 100
```
Comment (yyforyongyu) [inline itest/tx_publisher_test.go]: let's do sth like this rather than defining an arbitrary amount? https://github.com/btcsuite/btcwallet/blob/a6865a37e0493df62a7d895a47269022fe62ed66/itest/tx_creator_test.go#L723-L725

maybe we can create sth on the `harness` such that it can always give us a dust amount

File: itest/tx_publisher_test.go [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+		err := w.CheckMempoolAcceptance(h.Context(), rejection.tx)
+		require.ErrorIs(
+			h, err, want, "unexpected verdict for %s",
+			rejection.name,
+		)
+
+		h.AssertTxNotInMempool(rejection.tx.TxHash())
+	}
+
+	// A refused transaction leaves the coin it named still spendable.
```
Comment (yyforyongyu) [inline itest/tx_publisher_test.go:197]: 👍

File: bwtest/README.md [hunk 3]
```diff
@@ -68,6 +68,37 @@ Funding is also available on its own through `(*HarnessTest).FundWallet` and
...
+
+`(*HarnessTest).SignSpend` authors and signs a transaction from a wallet's own
+coins. It takes a `SpendFixture` naming the inputs, the outputs and the input
+sequence, and returns the signed transaction:
+
+```go
+tx := t.SignSpend(w, bwtest.SpendFixture{
+	Inputs:   []wire.OutPoint{funding.WalletOutpoints[0]},
+	Outputs:  []wire.TxOut{{Value: amount, PkScript: pkScript}},
+	Sequence: wire.MaxTxInSequenceNum,
```
Comment (yyforyongyu) [inline bwtest/README.md]: there is no such field?

**Review by yyforyongyu — CHANGES_REQUESTED** [review]
Again very easy to follow!

File: itest/tx_publisher_test.go [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+	// And the coin it named is untouched.
+	utxo, err := w.GetUtxo(h.Context(), spent)
+	require.NoError(h, err, "refused transaction consumed the coin")
+	require.True(h, utxo.Spendable, "coin is no longer spendable")
+}
+
+// testPublishWalletState verifies the lifecycle gate both publication methods
+// share: neither is available before the wallet starts or after it stops, and a
+// running wallet refuses a nil transaction rather than acting on it.
+func testPublishWalletState(h *bwtest.HarnessTest) {
```
Comment (yyforyongyu) [inline itest/tx_publisher_test.go]: This case covers both `CheckMempoolAcceptance` and `Broadcast`, across unstarted, running, and stopped states, and it also tests nil arguments. That combines several public contracts and interleaves multiple Act/Assert phases. Can we split this by API and behavior? For Task 435, separate stopped-wallet cases may be sufficient; the pre-start and nil checks could remain unit coverage or become distinct integration cases.

File: bwtest/backend.go [hunk 5]
```diff
@@ -46,6 +46,17 @@ type ChainBackend interface {
...
 }
 
+// MempoolAcceptanceSupported reports whether the chain backend under test
+// answers the mempool acceptance test.
+//
+// Neutrino is a light client with no mempool, so it reports the check as
+// unimplemented instead of returning a verdict. Publication test cases assert
+// an equally definite, but different, result against it, and this is the single
+// place that decides which backends have a mempool.
+func (h *HarnessTest) MempoolAcceptanceSupported() bool {
```
Comment (yyforyongyu) [inline bwtest/backend.go]: This derives a capability from the backend name even though the actual behavior is owned by the chain implementation, so the two can drift. The tests also accept `ErrUnimplemented` for neutrino, but the public `TxPublisher.CheckMempoolAcceptance` documentation does not describe that backend-dependent contract. Could we expose the capability through the backend contract and document the possible error on the public API?

File: bwtest/harness_wallet.go [hunk 6]
```diff
@@ -330,6 +332,93 @@ func (h *HarnessTest) FundWalletOfType(w *wallet.Wallet,
...
 }
 
+// DustThreshold returns the smallest amount an output paying pkScript may
+// carry without the network refusing it as dust. Any amount below it is dust.
+//
+// The limit depends on the output's script, so it is derived from the relay
+// policy itself rather than written down as a number that would silently drift
+// from it.
+func (h *HarnessTest) DustThreshold(pkScript []byte) btcutil.Amount {
+	h.Helper()
```
Comment (GustavoStingelin) [inline bwtest/harness_wallet.go:348]: Could this use `mempool.GetDustThreshold(&output)` directly? For the spendable scripts used by these tests, it appears to return the same threshold at `DefaultRelayFeePerKb`. If this loop handles another policy case, please document it.

File: bwtest/neutrino.go [hunk 7]
```diff
@@ -32,6 +32,15 @@ func NewNeutrinoBackend(t *testing.T, _ string) *NeutrinoBackend {
...
 }
 
+// SupportsMempoolAcceptance reports that neutrino does not answer the mempool
+// acceptance test. A light client keeps no mempool, so its TestMempoolAccept
+// returns chain.ErrUnimplemented for every transaction.
+//
+// NOTE: This is part of the ChainBackend interface.
+func (n *NeutrinoBackend) SupportsMempoolAcceptance() bool {
+	return false
+}
```
Comment (GustavoStingelin) [inline bwtest/neutrino.go:42]: 👍🏼

File: bwtest/harness_wallet.go [hunk 8]
```diff
@@ -330,6 +332,93 @@ func (h *HarnessTest) FundWalletOfType(w *wallet.Wallet,
...
+
+	// The wallet supplies each input's previous output and derivation path
+	// from its own records, so the fixture only has to name the outpoints.
+	// Unknown inputs are an error rather than a silently unsigned input,
+	// which would surface later as an opaque finalization failure.
+	packet, err = w.DecorateInputs(h.Context(), packet, false)
+	require.NoError(h, err, "failed to decorate psbt inputs")
+
+	err = w.FinalizePsbt(h.Context(), packet)
+	require.NoError(h, err, "failed to finalize psbt")
```
Comment (GustavoStingelin) [inline bwtest/harness_wallet.go:419]: Non-blocking: `FinalizePsbt` also signs wallet-owned inputs before finalization. A short comment here could clarify why no separate `SignPsbt` call is needed.

File: itest/tx_publisher_test.go (hunk 1 above)
Comment (Abdulkbk) [reply itest/tx_publisher_test.go]: Added to harness bc9fe4b7292ed9dbcdb6b2c7e6f70a37e653d261

File: bwtest/README.md (hunk 3 above)
Comment (Abdulkbk) [reply bwtest/README.md]: Stale doc, now fixed

File: itest/tx_publisher_test.go (hunk 4 above)
Comment (Abdulkbk) [reply itest/tx_publisher_test.go]: splitted

File: bwtest/backend.go (hunk 5 above)
Comment (Abdulkbk) [reply bwtest/backend.go]: Done, exposed it through backend contract and also document the public API

File: bwtest/harness_wallet.go [hunk 9]
```diff
@@ -330,6 +332,93 @@ func (h *HarnessTest) FundWalletOfType(w *wallet.Wallet,
...
+// carry without the network refusing it as dust. Any amount below it is dust.
+//
+// The limit depends on the output's script, so it is derived from the relay
+// policy itself rather than written down as a number that would silently drift
+// from it.
+func (h *HarnessTest) DustThreshold(pkScript []byte) btcutil.Amount {
+	h.Helper()
+
+	output := wire.TxOut{PkScript: pkScript}
+	for output.Value = 1; ; output.Value++ {
```
Comment (yyforyongyu) [inline bwtest/harness_wallet.go]: we can just do `return btcutil.Amount(mempool.GetDustThreshold(txOut))`

**Review by yyforyongyu — CHANGES_REQUESTED** [review]
One last thing, otherwise good to go!

**Review by GustavoStingelin — CHANGES_REQUESTED** [review]
looking very good, left just a few comments

File: bwtest/harness_wallet.go (hunk 9 above)
Comment (GustavoStingelin) [reply bwtest/harness_wallet.go]: left this one too, haha

Comment (Abdulkbk) [reply bwtest/harness_wallet.go]: done

File: bwtest/harness_wallet.go (hunk 6 above)
Comment (Abdulkbk) [reply bwtest/harness_wallet.go:348]: done

File: bwtest/harness_wallet.go (hunk 8 above)
Comment (Abdulkbk) [reply bwtest/harness_wallet.go:419]: added

File: bwtest/harness_wallet.go [hunk 10]
```diff
@@ -330,6 +332,98 @@ func (h *HarnessTest) FundWalletOfType(w *wallet.Wallet,
...
+// fee. A case paying a different relay fee cannot use this.
+//
+// The script is assumed spendable, which is what paying a wallet address
+// produces. An unspendable script is dust at any amount and has no threshold.
+func (h *HarnessTest) DustThreshold(pkScript []byte) btcutil.Amount {
+	h.Helper()
+
+	output := wire.TxOut{PkScript: pkScript}
+
+	return btcutil.Amount(mempool.GetDustThreshold(&output))
```
Comment (GustavoStingelin) [inline bwtest/harness_wallet.go:352]: Is it worth removing this function now that it's so simplified? Creating a whole function just to save one line of code might be overkill.

**Review by yyforyongyu — APPROVED** [review]
LGTM!

File: bwtest/harness_wallet.go (hunk 10 above)
Comment (yyforyongyu) [reply bwtest/harness_wallet.go:352]: I think it's fine - also depends on how often it's used as 2x reducation is good. But more importantly, I think it's there such that the agent won't miss using the correct way to calculate dust, which previously it ended up using a `for` loop

Comment (Abdulkbk) [reply bwtest/harness_wallet.go:352]: Yes, I think we should leave it too.

Comment (GustavoStingelin) [reply bwtest/harness_wallet.go:352]: ok

**Review by GustavoStingelin — APPROVED** [review]

**Comment by GustavoStingelin** [comment]
@Abdulkbk my rebase from ui removed ur sigs, could u rebase this one?
